### DISCUSSION TITLE
Is it possible to update the UI "in place" ?

### DISCUSSION THREAD

**Discussion by wsw70** [windmill-labs/windmill#2704] [Questions]
As I am progressing through Windmill (and, honestly, it is good) I am trying to understand how to make use of the UI.

I used basic input data and pushed them downstream in the workflow and now would like to provide the user with some feedback. There are two major cases I would like to harness the ability:

-  to add data in one input, and have another one updated through a background call (for instance, giving the ID of a person would look up the IU via an API and fill in a "details" entry)
- to send feedback to a user once his "submit" action was registered (it does not have to be dependent on the rest of the workflow, rather just have a visual indication that the button was successfully pressed)

Is this possible, and if so where should I look? (just a pointer to the rough place would be great - I am navigating through the docs but it takes some time to get to know what is where)

**Comment by rubenfiszel** [comment]
There are two ways to modify the UI in place:

Use a frontend script to modify the frontend state:
```
state.foo = 42
```

it will modify state.foo and anything that depends on state.foo will be re-evaluated

and the function
```
setValue('<id>', 42)
```

which can be used to set inputs of various components: https://www.windmill.dev/docs/apps/app-runnable-panel#setvalue

Is that what you were looking for ?
